client/web/src/cody/components/ChatEditor.ts

```diff
@@ -0,0 +1,141 @@
+import { URI } from 'vscode-uri'
+
+import {
+    ActiveTextEditor,
+    ActiveTextEditorSelection,
+    ActiveTextEditorVisibleContent,
+    Editor,
+} from '@sourcegraph/cody-shared/dist/editor'
+
+export interface EditorStore {
+    selectedText: string
+    content: string
+    filename: string
+    repo: string
+    revision: string
+}
+
+export class ChatEditor implements Editor {
+    private editor?: EditorStore | null
+    constructor(editor?: EditorStore | null) {
+        this.editor = editor
+    }
+
+    public get fileName(): string {
+        return this.editor?.filename || ''
+    }
+
+    public get repoName(): string | undefined {
+        return this.editor?.repo
+    }
+
+    public get revision(): string | undefined {
+        return this.editor?.revision
+    }
+
+    public getWorkspaceRootPath(): string | null {
+        return null
+    }
+
+    public getActiveTextEditor(): ActiveTextEditor | null {
+        const editor = this.editor
+        if (editor) {
+            return {
+                content: editor.content,
+                filePath: editor.filename,
+                repoName: this.repoName,
+                revision: this.revision,
+            }
+        }
+        return null
+    }
+
+    public getActiveTextEditorSelection(): ActiveTextEditorSelection | null {
+        const editor = this.editor
+
+        if (!editor?.content) {
+            return null
+        }
+
+        const splitText = editor.content.split(editor.selectedText)
+        const precedingText = splitText[0]
+        splitText.shift()
```

Comment:
Using destructuring assignment may improve readability in this case IMO.
```suggestion
        const [precedingText, ...splitText] = editor.content.split(editor.selectedText)
```

Reply:
I tried this but it caused a bug for the `followingText` property, so I'm keeping it as is for now